test/augment_test/csv/model/measures.codable.dart

```diff
@@ -0,0 +1,101 @@
+part of 'measures.dart';
+
+augment class Measures implements SelfEncodable {
+  const Measures(this.id, this.name, this.age, this.isActive, this.signupDate, this.website);
+
+  static const Codable<Measures> codable = MeasuresCodable();
+
+  @override
+  void encode(Encoder encoder) {
+    final keyed = encoder.encodeKeyed();
+    keyed.encodeString('id', id);
+    keyed.encodeStringOrNull('name', name);
+    keyed.encodeInt('age', age);
+    keyed.encodeBool('isActive', isActive);
+    keyed.encodeObjectOrNull('signupDate', signupDate, using: const DateTimeCodable());
+    keyed.encodeObjectOrNull('website', website, using: const UriCodable());
+    keyed.end();
+  }
+}
+
+// equatable augmentation
+augment class Measures {
+  @override
+  bool operator ==(Object other) {
+    return identical(this, other) ||
+        other is Measures &&
+            runtimeType == other.runtimeType &&
+            id == other.id &&
+            name == other.name &&
+            age == other.age &&
+            isActive == other.isActive &&
+            signupDate == other.signupDate &&
+            website == other.website;
+  }
+
+  @override
+  int get hashCode => Object.hash(id, name, age, isActive, signupDate, website);
+}
+
+// toString augmentation
+augment class Measures {
+  @override
+  String toString() {
+    return 'Measures{id: $id, name: $name, age: $age, isActive: $isActive, signupDate: $signupDate, website: $website}';
+  }
+}
+
+
+class MeasuresCodable extends SelfCodable<Measures> {
+  const MeasuresCodable();
+
+  @override
+  Measures decode(Decoder decoder) {
+    return switch (decoder.whatsNext()) {
+      DecodingType.mapped || DecodingType.map => decodeMapped(decoder.decodeMapped()),
+      DecodingType.keyed || DecodingType.unknown => decodeKeyed(decoder.decodeKeyed()),
+      _ => decoder.expect('keyed or mapped'),
+    };
+  }
+
+  Measures decodeKeyed(KeyedDecoder decoder) {
+    late String id;
+    late String? name;
+    late int age;
+    late bool isActive;
+    late DateTime? signupDate;
```

Comment:
Nullable options should not be late, since they may simply not appear at all.

Reply:
Good catch @TekExplorer !  In the case of nullable fields within the model the code generator should not  generate `late` modifiers because the runtime check for initialization (forced by using `late`) would generate an exception when the field had not been initialized explicitly to null.  
This block should look like this  
```
    late String id;
    String? name;
    late int age;
    late bool isActive;
    DateTime? signupDate;
    Uri? website;
```
So name and signupDate auto initialize to null (and could possible be changed to something else during the decode).
@schultek Thoughts ?